File: client_pizzeria/lib/infrastructure/mappers/pizza_mapper.dart
import 'package:client_pizzeria/domain/entities/pizza.dart';

class PizzaMapper {
  // Mètode estàtic que rep un JSON i retorna una instància de Pizza
  static Pizza fromJson(Map<String, dynamic> json) {
    return Pizza(
      id: json["id"],
      nom: json["nom"],
      desc: json["desc"],
      vegetariana: json["vegetariana"],
      img: json["img"],
      preu: json["preu"],
      alergens:
          json["alergens"] != null ? List<String>.from(json["alergens"]) : null,
    );
  }
}

File: server_pizzeria/src/app/infraestructure/mappers/PizzaMapper.ts
import Pizza from "../../domain/entities/Pizza.js";

// Mapper a JSON
export default class PizzaMapper {
    
    static toJSON(pizza: Pizza): object {
        return {
            id: pizza.id,
            nom: pizza.nom,
            desc: pizza.desc,
            vegetariana: pizza.vegetariana,
            alergens: pizza.alergens,
            img: pizza.img,
            preu: pizza.preu
        };
    }
}
